internal/app/singularity/cache_list_linux.go

```diff
@@ -31,82 +34,98 @@ func findSize(size int64) (string, error) {
+// and an error if one occures.
+func listLibraryCache(listFiles bool) (int, int64, error) {
+	var totalSize int64
+	count := 0
+
 	libraryCacheFiles, err := ioutil.ReadDir(cache.Library())
 	if err != nil {
-		return fmt.Errorf("unable to open library cache folder: %v", err)
+		return 0, 0, fmt.Errorf("unable to open library cache folder: %v", err)
 	}
 	for _, f := range libraryCacheFiles {
 		cont, err := ioutil.ReadDir(filepath.Join(cache.Library(), f.Name()))
 		if err != nil {
-			return fmt.Errorf("unable to look in library cache: %v", err)
+			return 0, 0, fmt.Errorf("unable to look in library cache: %v", err)
 		}
 		for _, c := range cont {
 			fileInfo, err := os.Stat(filepath.Join(cache.Library(), f.Name(), c.Name()))
 			if err != nil {
-				return fmt.Errorf("unable to get stat for library cache: %v", err)
+				return 0, 0, fmt.Errorf("unable to get stat for library cache: %v", err)
 			}
 			printFileSize, err := findSize(fileInfo.Size())
 			if err != nil {
 				// no need to describe the error, since it is already
 				sylog.Warningf("%v", err)
 			}
-			fmt.Printf("%-22s %-22s %-16s %s\n", c.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), printFileSize, "library")
+			if !listFiles {
+				fmt.Printf("%-22s %-22s %-16s %s\n", c.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), printFileSize, "library")
+			}
+			count++
+			totalSize += fileInfo.Size()
 		}
 	}
-	return nil
+	return count, totalSize, nil
 }
 
-func listOciCache() error {
-	// loop through oci-tmp cache
+// listOciCache will list all you oci-tmp cache.
+func listOciCache(listFiles bool) (int, int64, error) {
+	var totalSize int64
+	count := 0
+
 	ociTmp, err := ioutil.ReadDir(cache.OciTemp())
 	if err != nil {
-		return fmt.Errorf("unable to open oci-tmp folder: %v", err)
+		return 0, 0, fmt.Errorf("unable to open oci-tmp folder: %v", err)
 	}
 	for _, f := range ociTmp {
 		blob, err := ioutil.ReadDir(filepath.Join(cache.OciTemp(), f.Name()))
 		if err != nil {
-			return fmt.Errorf("unable to look in oci-tmp cache: %v", err)
+			return 0, 0, fmt.Errorf("unable to look in oci-tmp cache: %v", err)
 		}
 		for _, b := range blob {
 			fileInfo, err := os.Stat(filepath.Join(cache.OciTemp(), f.Name(), b.Name()))
 			if err != nil {
-				return fmt.Errorf("unable to get stat for oci-tmp cache: %v", err)
+				return 0, 0, fmt.Errorf("unable to get stat for oci-tmp cache: %v", err)
 			}
 			printFileSize, err := findSize(fileInfo.Size())
 			if err != nil {
 				// no need to describe the error, since it is already
 				sylog.Warningf("%v", err)
 			}
-			fmt.Printf("%-22s %-22s %-16s %s\n", b.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), printFileSize, "oci")
+			if !listFiles {
+				fmt.Printf("%-22s %-22s %-16s %s\n", b.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), printFileSize, "oci")
+			}
+			count++
+			totalSize += fileInfo.Size()
 		}
 	}
-	return nil
+	return count, totalSize, nil
 }
```

Comment:
looks good, although the listOcicache, listLibraryCache and listBlobCache functions are very similar, would be nice to have a single listCache function handling the logic by passing in there a param to define whether it's blob, library or oci , this may need additional changes since these functions are called on other places but might be good to have it on mind for later on :+1:

Reply:
I combined `listLibraryCache` and `listOciCache` into one, but `listOciCache` is a little different, thats still its own function.